airsonic-main/src/main/java/org/airsonic/player/ajax/PlayQueueService.java

```diff
@@ -645,20 +684,43 @@ private PlayQueueInfo convert(HttpServletRequest request, Player player, boolean
     }
 
     private PlayQueueInfo convert(HttpServletRequest request, Player player, boolean serverSidePlaylist, int offset) throws Exception {
+
         String url = NetworkService.getBaseUrl(request);
+        Locale locale = RequestContextUtils.getLocale(request);
+        PlayQueue playQueue = player.getPlayQueue();
 
-        /* if (serverSidePlaylist && player.isJukebox()) {
-            updateJukebox(player, offset);
-        } */
-        boolean isCurrentPlayer = player.getIpAddress() != null && player.getIpAddress().equals(request.getRemoteAddr());
+        List<PlayQueueInfo.Entry> entries;
+        if (playQueue.isInternetRadioEnabled()) {
+            entries = convertInternetRadio(request, player);
+        } else {
+            entries = convertMediaFileList(request, player);
+        }
 
+        boolean isCurrentPlayer = player.getIpAddress() != null && player.getIpAddress().equals(request.getRemoteAddr());
+        boolean isStopEnabled = playQueue.getStatus() == PlayQueue.Status.PLAYING && !player.isExternalWithPlaylist();
         boolean m3uSupported = player.isExternal() || player.isExternalWithPlaylist();
         serverSidePlaylist = player.isAutoControlEnabled() && m3uSupported && isCurrentPlayer && serverSidePlaylist;
-        Locale locale = RequestContextUtils.getLocale(request);
 
-        List<PlayQueueInfo.Entry> entries = new ArrayList<PlayQueueInfo.Entry>();
+        float gain = jukeboxService.getGain(player);
+
+        return new PlayQueueInfo(
+            entries,
+            isStopEnabled,
+            playQueue.isRepeatEnabled(),
+            playQueue.isShuffleRadioEnabled(),
+            playQueue.isInternetRadioEnabled(),
+            serverSidePlaylist,
+            gain
+        );
+    }
+
+    private List<PlayQueueInfo.Entry> convertMediaFileList(HttpServletRequest request, Player player) {
+
+        String url = NetworkService.getBaseUrl(request);
+        Locale locale = RequestContextUtils.getLocale(request);
         PlayQueue playQueue = player.getPlayQueue();
 
+        List<PlayQueueInfo.Entry> entries = new ArrayList<PlayQueueInfo.Entry>();
```

Comment:
No need to specify the type on the right side, using the diamond operator (`<>`) is the recommended way in *modern* Java.

Reply:
Good to know, thanks!